engine/src/flutter/lib/web_ui/lib/src/engine/semantics/scrollable.dart

```diff
@@ -234,20 +234,16 @@ class SemanticScrollable extends SemanticRole {
         // "scroll" because it works both on Android and iOS.
         if (semanticsObject.isVerticalScrollContainer) {
           element.style.overflowY = 'scroll';
-        } else {
-          assert(semanticsObject.isHorizontalScrollContainer);
+        } else if (semanticsObject.isHorizontalScrollContainer) {
           element.style.overflowX = 'scroll';
+        } else {
+          element.style.overflow = 'hidden';
         }
       case GestureMode.pointerEvents:
         // We use "hidden" instead of "scroll" so that the browser does
         // not "steal" pointer events. Flutter gesture recognizers need
         // all pointer events in order to recognize gestures correctly.
-        if (semanticsObject.isVerticalScrollContainer) {
-          element.style.overflowY = 'hidden';
-        } else {
-          assert(semanticsObject.isHorizontalScrollContainer);
-          element.style.overflowX = 'hidden';
-        }
+        element.style.overflow = 'hidden';
```

Comment:
When gesture mode flips from browser to pointer events, this will result in `overflowY == 'scroll'` and `overflow == 'hidden'`. Is that a good state to be in?

The vice versa situation (flipping back to browser gesture mode) seems equally questionable.

Reply:
Tried in a jsfiddle, the following:
```javascript
element.style.overflow = 'hidden';
element.style.overflowY = 'scroll';
```
it results in the element having `overflowY: scroll` and `overflowX: hidden`.

But in order to avoid this whole situation, we can just always do:
```dart
element.style.removeProperty('overflow');
```
whenever the gesture mode is changed. WDYT?

Reply:
Done!

Reply:
In that case, I think when we're setting `overflow` we should remove `overflowX/Y`?